metricbeat/module/docker/cpu/helper.go

```diff
@@ -83,24 +83,24 @@ func (c *CPUService) PerCpuUsage(stats *dc.Stats) common.MapStr {
 	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOLdCpu(stats).PerCpuUsage) {
 		output = common.MapStr{}
 		for index := range getNewCpu(stats).PerCpuUsage {
-			output[strconv.Itoa(index)] = c.calculateLoad(getNewCpu(stats).PerCpuUsage[index] - getOLdCpu(stats).PerCpuUsage[index])
+			output[strconv.Itoa(index)] = c.calculateLoad(int64(getNewCpu(stats).PerCpuUsage[index] - getOLdCpu(stats).PerCpuUsage[index]))
 		}
 	}
 	return output
 }
 
 func (c *CPUService) TotalUsage(stats *dc.Stats) float64 {
-	return c.calculateLoad(getNewCpu(stats).TotalUsage - getOLdCpu(stats).TotalUsage)
+	return c.calculateLoad(int64(getNewCpu(stats).TotalUsage - getOLdCpu(stats).TotalUsage))
 }
 
 func (c *CPUService) UsageInKernelmode(stats *dc.Stats) float64 {
-	return c.calculateLoad(getNewCpu(stats).UsageInKernelmode - getOLdCpu(stats).UsageInKernelmode)
+	return c.calculateLoad(int64(getNewCpu(stats).UsageInKernelmode - getOLdCpu(stats).UsageInKernelmode))
 }
 
 func (c *CPUService) UsageInUsermode(stats *dc.Stats) float64 {
-	return c.calculateLoad(getNewCpu(stats).UsageInUsermode - getOLdCpu(stats).UsageInUsermode)
+	return c.calculateLoad(int64(getNewCpu(stats).UsageInUsermode - getOLdCpu(stats).UsageInUsermode))
 }
 
-func (c *CPUService) calculateLoad(value uint64) float64 {
+func (c *CPUService) calculateLoad(value int64) float64 {
```

Comment:
What does this function do? The name doesn't seem to match up with what I expected. It looks more like it is converting the value given in ns to seconds? If that is true, are you sure the `value` is actually in nanoseconds? 

Normally the units on these values are in jiffies, but maybe Docker is converting them from jiffies to nanoseconds based on the `_SC_CLK_TCK` value from the OS.